### DISCUSSION TITLE
Refuse an incoming message?

### DISCUSSION THREAD

**Discussion by GitHK** [ag2ai/faststream#1371] [Q&A] [answered]
In a subscriber, is there a way to refuse the incoming message? I would like for the message to be automatically put at the beginning of the queue.

I need to implement a worker that if it's busy it should not pickup any message from the queue. how do I achieve this?

Thanks for the help!

**Comment by Lancetnik** [comment, marked as answer]
About "refusing" you should search for decision in [Acknowledgement](https://faststream.airt.ai/latest/rabbit/ack/) your broker section. It is all about your broker behavior, **FastStream** just provides you with suitable API.

About blocking consuming: some broker like Kafka/Nats are working this way already. You can also take a look at your broker mechanism (like max qos for RMQ) or just use `asyncio.Lock`.

**Reply by GitHK** [reply]
Thanks a lot. Indeed the Acknowledgement word was what I was searching for. This helps me with my problem.
